packages/caver-account/src/accountKey/accountKeyRoleBased.js

```diff
@@ -0,0 +1,125 @@
+            throw new Error(`Cannot decode to AccountKeyRoleBased: ${rlpEncodedKey}`)
+
+        const keys = RLP.decode(`0x${rlpEncodedKey.slice(ACCOUNT_KEY_TAG.ACCOUNT_KEY_ROLE_BASED_TAG.length)}`)
+        const accountKeys = []
+        for (const key of keys) {
+            if (key.startsWith(ACCOUNT_KEY_TAG.ACCOUNT_KEY_ROLE_BASED_TAG)) throw new Error('Nested role based key.')
+            if (key.startsWith(ACCOUNT_KEY_TAG.ACCOUNT_KEY_NIL_TAG)) {
+                accountKeys.push(undefined)
+            } else if (key.startsWith(ACCOUNT_KEY_TAG.ACCOUNT_KEY_LEGACY_TAG)) {
+                accountKeys.push(AccountKeyLegacy.decode(key))
+            } else if (key.startsWith(ACCOUNT_KEY_TAG.ACCOUNT_KEY_PUBLIC_TAG)) {
+                accountKeys.push(AccountKeyPublic.decode(key))
+            } else if (key.startsWith(ACCOUNT_KEY_TAG.ACCOUNT_KEY_FAIL_TAG)) {
+                accountKeys.push(AccountKeyFail.decode(key))
+            } else if (key.startsWith(ACCOUNT_KEY_TAG.ACCOUNT_KEY_WEIGHTED_MULTISIG_TAG)) {
+                accountKeys.push(AccountKeyWeightedMultiSig.decode(key))
+            } else {
+                throw new Error(`Failed to decode RLP-encoded account key. Invalid RLP-encoded account key ${key}`)
+            }
+        }
+        return new AccountKeyRoleBased(accountKeys)
+    }
+
+    /**
+     * Creates an instance of AccountKeyRoleBased.
+     * @param {Array.<Array.<string>>} roleBasedPubArray - An array of public key strings.
+     * @param {Array.<object>} options - An array of options which defines threshold and weight.
+     * @return {AccountKeyRoleBased}
+     */
+    static fromRoleBasedPublicKeysAndOptions(roleBasedPubArray, options) {
+        const accountKeys = []
+        // Format will be like below
+        // publicKeyArray = [[pub, pub], [pub], [pub, pub, pub]]
+        // options = [{threshold: 1, weight: [1,1]}, {}, {threshold: 1, weight: [1,1,1]}]
+        for (let i = 0; i < roleBasedPubArray.length; i++) {
+            if (roleBasedPubArray[i].length > 0) {
+                if (roleBasedPubArray[i].length === 1 && Object.keys(options[i]).length === 0) {
+                    accountKeys.push(new AccountKeyPublic(roleBasedPubArray[i][0]))
+                } else {
+                    accountKeys.push(AccountKeyWeightedMultiSig.fromPublicKeysAndOptions(roleBasedPubArray[i], options[i]))
+                }
+            } else {
+                accountKeys.push(undefined)
+            }
+        }
+        return new AccountKeyRoleBased(accountKeys)
+    }
+
+    /**
+     * Create an instance of AccountKeyRoleBased.
+     * @param {Array.<Array.<AccountKeyLegacy|AccountKeyPublic|AccountKeyFail|AccountKeyWeightedMultiSig>>} accountKeyArray - A two-dimensional array containing arrays of instances of AccountKeyPublic or AccountKeyWeightedMultiSig for each role.
+     */
+    constructor(accountKeyArray) {
+        this._accountKeys = accountKeyArray
```

Comment:
Don't we need to check the type of `accountKeyArray`?

Reply:
I added format validation :)